pkg/manager/member/pump_member_manager_test.go

```diff
@@ -715,3 +718,60 @@ func TestGetNewPumpConfigMap(t *testing.T) {
 }
 
 // TODO: add ut for getPumpStatefulSet
+func TestSyncPumpStatefulSetForTidbCluster(t *testing.T) {
+	g := NewGomegaWithT(t)
+	type testcase struct {
+		name        string
+		updateTC    func(*v1alpha1.TidbCluster)
+		upgradingFn func(corelisters.PodLister, *appsv1.StatefulSet, *v1alpha1.TidbCluster) (bool, error)
+		errExpectFn func(*GomegaWithT, error)
+		tcExpectFn  func(*GomegaWithT, *v1alpha1.TidbCluster)
+	}
+	status := appsv1.StatefulSetStatus{
+		Replicas: int32(3),
+	}
+	testFn := func(test *testcase, t *testing.T) {
+		tc := newTidbClusterForPump()
+
+		set := &appsv1.StatefulSet{
+			Status: status,
+		}
+		if test.updateTC != nil {
+			test.updateTC(tc)
+
+		}
+		pmm, _, _ := newFakePumpMemberManager()
+
+		err := pmm.syncTiDBClusterStatus(tc, set)
+
+		if test.errExpectFn != nil {
+			test.errExpectFn(g, err)
+		}
+		if test.tcExpectFn != nil {
+			test.tcExpectFn(g, tc)
+		}
+	}
+	tests := []testcase{
+		{
+			name: "statefulset is upgrading",
+			updateTC: func(tc *v1alpha1.TidbCluster) {
```

Comment:
instead of `updateTC`, we should change the statefulset to mimic an update, because the status of tc is taken from statefulset